src/seedu/addressbook/commands/Command.java

```diff
@@ -26,6 +26,7 @@ public Command(int targetIndex) {
     protected Command() {
     }
 
+    public abstract boolean isMutating();
```

Comment:
This method should have a header comment. If not, users of the method as well as those who overrides method will not know how exactly the method behaves.